src/System.Data.SqlClient/src/System/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1900,6 +1932,33 @@ internal void OnFeatureExtAck(int featureId, byte[] data)
                         }
                         break;
                     }
+                case TdsEnums.FEATUREEXT_FEDAUTH:
+                    {
+                        if (!_federatedAuthenticationRequested)
+                        {
+                            throw SQL.ParsingErrorFeatureId(ParsingErrorState.UnrequestedFeatureAckReceived, featureId);
+                        }
+
+                        Debug.Assert(_fedAuthFeatureExtensionData != null, "_fedAuthFeatureExtensionData must not be null when _federatedAuthenticatonRequested == true");
+
+                        switch (_fedAuthFeatureExtensionData.Value.libraryType)
+                        {
+                            case TdsEnums.FedAuthLibrary.ADAL: // For later support
```

Comment:
Is the server, ever expected to send `TdsEnums.FedAuthLibrary.ADAL` for AccessToken?